src/databases/WData/avtwdataFileFormat.C

```diff
@@ -0,0 +1,861 @@
+            varunit.push_back(md->var[vid].unit);
+            trans.push_back(cabs);
+
+            varlabel = md->link[vid].name;
+            varlabel += "_arg";
+            varname.push_back(varlabel);
+            varunit.push_back("PI");
+            trans.push_back(carg);
+
+            varlabel = md->link[vid].name;
+            varlabel += "_re";
+            varname.push_back(varlabel);
+            varunit.push_back(md->var[vid].unit);
+            trans.push_back(cre);
+
+            varlabel = md->link[vid].name;
+            varlabel += "_im";
+            varname.push_back(varlabel);
+            varunit.push_back(md->var[vid].unit);
+            trans.push_back(cim);
+        }
+}
+
+bool WDataComplexVariable::getVariable(const char *_varname, int cycleid, float *data_for_visit)
+{
+    // check it this varaible can generate data for varname
+    int varid = -1;
+
+    for (int i = 0; i < varname.size(); i++)
+        if (strcmp(_varname, varname[i].c_str()) == 0)
+            varid = i;
+
+    if (varid == -1)
+        return false;
+
+    // load data
+    loadCycle(cycleid);
+
+    int ix, iy, iz;
+    int ixyz1 = 0, ixyz2 = 0;
+    if (trans[varid] == cabs)
+    {
+        for (ix = 0; ix < md->nx; ix++)
+            for (iy = 0; iy < md->ny; iy++)
+                for (iz = 0; iz < md->nz; iz++)
+                {
+                    ixyz2 = ix + md->nx * iy + md->nx * md->ny * iz;
+                    data_for_visit[ixyz2] = (float)abs(dataC[ixyz1]);
+
+                    ixyz1++;
+                }
+    }
+    else if (trans[varid] == carg)
+    {
+        for (ix = 0; ix < md->nx; ix++)
+            for (iy = 0; iy < md->ny; iy++)
+                for (iz = 0; iz < md->nz; iz++)
+                {
+                    ixyz2 = ix + md->nx * iy + md->nx * md->ny * iz;
+                    data_for_visit[ixyz2] = (float)(arg(dataC[ixyz1]) / M_PI);
+
+                    ixyz1++;
+                }
+    }
+    else if (trans[varid] == cre)
+    {
+        for (ix = 0; ix < md->nx; ix++)
+            for (iy = 0; iy < md->ny; iy++)
+                for (iz = 0; iz < md->nz; iz++)
+                {
+                    ixyz2 = ix + md->nx * iy + md->nx * md->ny * iz;
+                    data_for_visit[ixyz2] = (float)(dataC[ixyz1].real());
+
+                    ixyz1++;
+                }
+    }
+    else if (trans[varid] == cim)
+    {
+        for (ix = 0; ix < md->nx; ix++)
+            for (iy = 0; iy < md->ny; iy++)
+                for (iz = 0; iz < md->nz; iz++)
+                {
+                    ixyz2 = ix + md->nx * iy + md->nx * md->ny * iz;
+                    data_for_visit[ixyz2] = (float)(dataC[ixyz1].imag());
+
+                    ixyz1++;
+                }
+    }
+
+    return true;
+}
+
+// =======================================================================================
+// ============================= WDataVectorVariable =======================================
```

Comment:
Style nitpick: There are 2 extra "=" at the end of the line.

Reply:
Done